test/e2e/framework/cloudevents/cloudevents.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2022 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package cloudevents contains helpers to interact with CloudEvents.
+package cloudevents
+
+import (
+	"context"
+	"encoding/json"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+	"github.com/cloudevents/sdk-go/v2/protocol/http"
+
+	"github.com/triggermesh/triggermesh/test/e2e/framework"
+)
+
+// E2ECeExtension is a CloudEvent extension that is used to propagate
+// the name of the test that generated the event.
+const E2ECeExtension = "iotriggermeshe2e"
+
+const (
+	curlContainerName     = "curl"
+	curlContainerImage    = "curlimages/curl:latest"
+	eventSenderNamePrefix = "eventsender-"
+)
+
+// NewEvent generates a CloudEvent with dummy values.
+func NewEvent(f *framework.Framework) *cloudevents.Event {
```

Comment:
This strikes me as more specific to the SQS target than something we can use in general.

Reply:
Why that? Literally every target needs to be tested by sending a CloudEvent to it, and this helper accepts an arbitrary event to make it versatile.

In fact, this helper comes directly from test-infra.

Reply:
Correction: my comment was related to `RunEventSender()` (oops).

Regardless, every target that isn't an "integration" target will happily accept an arbitrary payload and store it (Elasticsearch, anything Kafka-based, Kinesis, SNS, SQS, S3, Event Hubs, Pub/Sub, ... the list goes on and on).

Reply:
I do see your point, and can give you the same number of SaaS specific integrations that are looking for a specific target to use. Perhaps renaming this as `NewHelloEvent` to differentiate between the two use cases. Especially since some targets such as the S3 target can take a generic event _and_ a specific event of `io.triggermesh.awss3.object.put` (as per https://github.com/triggermesh/triggermesh/blob/bd69609b0c5d45a2a5c71f72239bc73d6e6904c5/pkg/apis/targets/v1alpha1/aws_s3_lifecycle.go#L27).

Reply:
I agree, `NewHelloEvent` is more descriptive.

The only purpose of this helper is really to avoid having to repeat `NewEvent ... SetType 
...` in every test when all we need is _some_ event. It's not meant to support _all_ target tests.